src/libs/kata-types/src/annotations/mod.rs

```diff
@@ -12,3 +23,799 @@ pub mod crio;
+/// A sandbox annotation that determines if pprof enabled.
+pub const KATA_ANNO_CONF_ENABLE_PPROF: &str = "io.katacontainers.config.runtime.enable_pprof";
+/// A sandbox annotation that determines if experimental features enabled.
+pub const KATA_ANNO_CONF_EXPERIMENTAL: &str = "io.katacontainers.config.runtime.experimental";
+/// A sandbox annotaion that determines how the VM should be connected to the the container network
+/// interface.
+pub const KATA_ANNO_CONF_INTER_NETWORK_MODEL: &str =
+    "io.katacontainers.config.runtime.internetworking_model";
+/// SandboxCgroupOnly is a sandbox annotation that determines if kata processes are managed only in sandbox cgroup.
+pub const KATA_ANNO_CONF_SANDBOX_CGROUP_ONLY: &str =
+    "io.katacontainers.config.runtime.sandbox_cgroup_only";
+/// A sandbox annotation that determines if create a netns for hypervisor process.
+pub const KATA_ANNO_CONF_DISABLE_NEW_NETNS: &str =
+    "io.katacontainers.config.runtime.disable_new_netns";
+/// A sandbox annotation to specify how attached VFIO devices should be treated.
+pub const KATA_ANNO_CONF_VFIO_MODE: &str = "io.katacontainers.config.runtime.vfio_mode";
+
+/// A helper structure to query configuration information by check annotations.
+#[derive(Debug, Default, Deserialize)]
+pub struct Annotation {
+    annotations: HashMap<String, String>,
+}
+
+impl From<HashMap<String, String>> for Annotation {
+    fn from(annotations: HashMap<String, String>) -> Self {
+        Annotation { annotations }
+    }
+}
+
+impl Into<HashMap<String, String>> for Annotation {
+    fn into(self) -> HashMap<String, String> {
+        self.annotations
+    }
+}
+
+impl Annotation {
+    /// Create a new instance of [`Annotation`].
+    pub fn new(annotations: HashMap<String, String>) -> Annotation {
+        Annotation { annotations }
+    }
+
+    /// Deserialize an object from a json string.
+    pub fn deserialize<T>(path: &str) -> Result<T>
+    where
+        for<'a> T: Deserialize<'a>,
+    {
+        let f = BufReader::new(File::open(path)?);
+        Ok(serde_json::from_reader(f)?)
+    }
+
+    /// Get an immutable reference to the annotation hashmap.
+    pub fn get_annotation(&self) -> &HashMap<String, String> {
+        &self.annotations
+    }
+
+    /// Get a mutable reference to the annotation hashmap.
+    pub fn get_annotation_mut(&mut self) -> &mut HashMap<String, String> {
+        &mut self.annotations
+    }
+
+    /// Get the value of annotation with `key` as string.
+    pub fn get(&self, key: &str) -> Option<String> {
+        let v = self.annotations.get(key)?;
+        let value = v.trim();
+
+        if !value.is_empty() {
+            Some(String::from(value))
+        } else {
+            None
+        }
+    }
+
+    /// Get the value of annotation with `key` as bool.
+    pub fn get_bool(&self, key: &str) -> Option<bool> {
+        if let Some(value) = self.get(key) {
+            let value = value.trim();
+            if value == "true" {
```

Comment:
[parse or from_str](https://doc.rust-lang.org/std/primitive.bool.html#method.from_str) is more ideal.

Reply:
it will be fixed in next version